docs/ingestion/schema-design.md

```diff
@@ -241,12 +240,45 @@ You should query for the number of ingested rows with:
 ]
 ```
 
-### Schema-less dimensions
+### Schema auto-discovery for dimensions
+
+Druid can infer the schema for your data in one of two ways:
+
+- [Type-aware schema discovery (experimental)](#type-aware-schema-discovery-experimental) where Druid infers the schema and type for your data. Type-aware schema discovery is experimental currently available for native batch and streaming ingestion.
+- [String-based schema discovery](#string-based-schema-discovery) where all the discovered columns are typed as either native string or multi-value string columns.
+
+#### Type-aware schema discovery (experimental)
```

Comment:
I think it is better not to put `(experimental)` in the heading itself as later, when we remove it we will have to remember to add a redirect. Better to just mention it in the warning text below

Reply:
```suggestion
#### Type-aware schema discovery
```